web_monitoring/diffing_server.py

```diff
@@ -325,35 +313,40 @@ def check_response_for_error(self, response):
 
 def _extract_encoding(headers, content):
     encoding = None
-    content_type = headers.get('Content-Type', '')
+    content_type = headers.get('Content-Type', '').lower()
     if 'charset=' in content_type:
         encoding = content_type.split('charset=')[-1]
-    if not encoding:
-        meta_tag_match = META_TAG_PATTERN.search(content, endpos=2048)
-        if meta_tag_match:
-            encoding = meta_tag_match.group(1).decode('ascii', errors='ignore')
-    if not encoding:
-        prolog_match = XML_PROLOG_PATTERN.search(content, endpos=2048)
-        if prolog_match:
-            encoding = prolog_match.group(1).decode('ascii', errors='ignore')
```

Comment:
We felt it was important to behave like a web browser does here, which is to look for these items, in this order, and obey them before trying to detect the encoding. CChardet doesn’t look at any of these, and my thinking was that if we added cChardet, we’d use it *in addition* to this logic for exactly this reason. (See this PR for more on that reasoning: https://github.com/edgi-govdata-archiving/web-monitoring-processing/pull/161)

Do you have a strong reason for why these should be removed? If not, I would like to make sure we keep them.

Reply:
I read #161. IMHO you depend too much on website defined encoding and they are wrong some times from my experience. 
I do not disagree strongly with your approach, I just thought correct encoding was covered with `cchardet` and we could also save some CPU cycles by removing the regex.
I will update my PR to keep them. NP.

Reply:
I hear you on this; it’s just that when I originally tried using cchardet the way you are in #161, I frequently saw *diffs display pages with questionable encoding differently than I would if I just viewed the page in my browser*, and I feel strongly that was a problem.

I’m all for not overspending on CPU cycles, and if you think there’s a more efficient way to match this logic than by using a regex (or if we can better optimize the regex), that’d be great.